src/main/java/efs/task/syntax/GuessNumberGame.java

```diff
@@ -12,11 +22,74 @@ public static void main(String[] args) {
         }
     }
 
-    public GuessNumberGame(String argument) {
-        //TODO: Implement the constructor
+    public GuessNumberGame(String argument) throws IllegalArgumentException {
+
+        try {
+            this.M = Integer.parseInt(argument);
+        } catch(NumberFormatException ex) {
+            System.out.println(UsefulConstants.WRONG_ARGUMENT);
+            throw new IllegalArgumentException();
+        }
+
+        if(M > 400 || M < 1) {
+            System.out.println(UsefulConstants.WRONG_ARGUMENT);
+            System.out.println("Jest spoza zakresu <1," + UsefulConstants.MAX_UPPER_BOUND + ">");
+            throw new IllegalArgumentException();
+        }
+
+        L = (int)(Math.floor(Math.log(M) / Math.log(2)) + 1);
+
+        numberToGuess = RandomGenerator.getDefault().nextInt(M) + 1;
     }
 
     public void play() {
-        //TODO: Implement the method that executes the game session
+
+        int yourAnswer;
+
+        System.out.println("Zagrajmy. Zgadnij liczbę z zakresu <1," + M + ">");
+
+        while(numberOfTries <= L) {
+
+            Scanner scanner = new Scanner(System.in);
+
+            printStateBar(numberOfTries, L);
+
+            System.out.println(UsefulConstants.GIVE_ME + " liczbę:");
+
+            try {
+                yourAnswer = scanner.nextInt();
+            } catch (Exception e) {
+                System.out.println(UsefulConstants.NOT_A_NUMBER);
+                numberOfTries++;
+                continue;
+            }
+
+            if(yourAnswer < numberToGuess) {
+                System.out.println(UsefulConstants.TO_LESS);
+            }
+            else if(yourAnswer > numberToGuess) {
+                System.out.println(UsefulConstants.TO_MUCH);
+            }
+            else {
+                System.out.println(UsefulConstants.YES);
+                System.out.println(UsefulConstants.CONGRATULATIONS + " " + numberOfTries + " - tyle prob zajelo Ci odgadniecie liczby " + numberToGuess);
+                break;
+            }
+            numberOfTries++;
+            scanner.close();
+        }
+        if(numberOfTries - 1 == L) {
+            System.out.println(UsefulConstants.UNFORTUNATELY + ", wyczerpales limit prob (" + L + ")" + " do ogadniecia liczby " + numberToGuess);
+        }
+    }
+    private void printStateBar(int numberOfTries, int limit) {
+        System.out.print("Twoje proby: [");
```

Comment:
_clean code_ można spróbować użyć w tej metodzie:
_https://www.geeksforgeeks.org/string-class-repeat-method-in-java-with-examples/_